### ISSUE TITLE
[Accessibility] An element of the given ControlType must support the Text pattern when the DataGridViewTextBoxCell is not in the editing state

### ISSUE THREAD

**Issue by Jessie-Zhang01** [dotnet/winforms#4031]
* .NET Core Version: .Net SDK 6.0.100-alpha.1.20479.7

* Have you experienced this same bug with .NET Framework?: 
Yes

**Minimal repro:**

1. Open the attached [DataGridViewTextBoxCell.zip](https://github.com/dotnet/winforms/files/5302789/DataGridViewTextBoxCell.zip) application in VS, then build and run it.
2. Open Accessibility Insights and test the Column1 in Form.

**Expected behavior:**
An element of the given ControlType support the Text pattern. The behavior should be like following screenshot from the DataGridViewTextBoxCell is in the editing state.
![image](https://user-images.githubusercontent.com/45864649/94635258-d4361a00-0304-11eb-9286-dfb871d7584d.png)

**Actual behavior:**
There is an error message: An element of the given ControlType must support the Text pattern. Following screenshot from the DataGridViewTextBoxCell is not in the editing state
![image](https://user-images.githubusercontent.com/45864649/94635271-dac49180-0304-11eb-8824-dc9cceb993d2.png)

**Comment by Jessie-Zhang01** [comment]
Take the following Guy Barker‘s suggestion as a reference from VSO bug: [1159872](https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1159872)
Hi all, by default I'd say that a TextBox should support the Text pattern when editable and when read-only.  The image below shows the UWP XAML Controls Gallery app, with Narrator positioned at a word in a read-only TextBox. A customer can use Capslock+J/K/L to move through the text word-by-word, or Capslock+M/,/. to move character-by-character, which can help with comprehension or spelling. 
![image](https://user-images.githubusercontent.com/45864649/94639955-f97c5580-030f-11eb-94bd-b0ab712bfddf.png)

Also, while the UWP XAML DataGrid in the Community Tooltip is inaccessible, the read-only TextBlock in the DataGrid does seem to support the Text pattern. The image below shows AIWin using the Text pattern to get the read-only text from the TextBlock, and moving through the text character-by-character.
![image](https://user-images.githubusercontent.com/45864649/94639974-01d49080-0310-11eb-9984-af8fa561797d.png)

**Comment by vladimir-krestov** [comment]
Closed #3909 as a duplicate. Take a look for more info

**Comment by merriemcgaw** [comment]
@vladimir-krestov can you guys put this higher on the priority list so we can close out all remaining bugs from the previous accessibility test pass? I've moved it to the todo column in the project.

**Comment by Zheng-Li01** [comment]
Verified the issue with latest 6.0.100-preview.7.21320.4 that the issue has been fixed, no error appeared when the DataGridViewTextBoxCell is not in the editing state, see below GIF.
![Issue4031](https://user-images.githubusercontent.com/38325459/122730407-c23e4200-d2ac-11eb-8957-af4d295f1632.gif)

**Comment by John-Qiao** [comment]
Verified the issue with .NET 6.0 6.0.100-preview.6.21323.1, it has been fixed as above testing result too.
